DigitalLearningSolutions.Web.Tests/Services/LoginServiceTests.cs

```diff
@@ -477,6 +499,81 @@ bool centreActive
+        {
+            // Given
+            var userEntity = new UserEntity(
+                UserTestHelper.GetDefaultUserAccount(),
+                new List<AdminAccount>(),
+                new List<DelegateAccount>
+                {
+                    UserTestHelper.GetDefaultDelegateAccount(),
+                    UserTestHelper.GetDefaultDelegateAccount(3, centreId: 2, candidateNumber: "AB2"),
+                }
+            );
+
+            var resultListingPrimaryEmailAsUnverified = ("primary@email.com",
+                new List<(int centreId, string centreName, string centreEmail)>());
+
+            A.CallTo(() => userService.GetUserByUsername(Username)).Returns(userEntity);
+            A.CallTo(() => userVerificationService.VerifyUserEntity(Password, userEntity))
+                .Returns(new UserEntityVerificationResult(true, new List<int>(), new[] { 2 }, new List<int>()));
+            A.CallTo(() => userService.GetUnverifiedEmailsForUser(userEntity.UserAccount.Id))
+                .Returns(resultListingPrimaryEmailAsUnverified);
+            A.CallTo(() => userService.ResetFailedLoginCount(userEntity.UserAccount)).DoesNothing();
+
+            // When
+            var result = loginService.AttemptLogin(Username, Password);
+
+            // Then
+            using (new AssertionScope())
+            {
+                result.UserEntity.Should().Be(userEntity);
+                result.LoginAttemptResult.Should().Be(LoginAttemptResult.UnverifiedEmail);
+                result.CentreToLogInto.Should().BeNull();
+            }
+        }
+
+        [Test]
+        public void
+            AttemptLogin_returns_unverified_email_if_user_is_logging_into_single_centre_and_centre_email_is_unverified()
+        {
+            // Given
+            const int centreId = 1;
+            var userEntity = new UserEntity(
+                UserTestHelper.GetDefaultUserAccount(),
+                new List<AdminAccount>(),
+                new List<DelegateAccount>
+                {
+                    UserTestHelper.GetDefaultDelegateAccount(centreId: 1),
+                }
+            );
+
+            var resultListingCentreEmailAsUnverified = ((string?)null,
+                new List<(int centreId, string centreName, string centreEmail)>
+                    { (centreId, "Test Centre", "centre@email.com") });
+
+            A.CallTo(() => userService.GetUserByUsername(Username)).Returns(userEntity);
+            A.CallTo(() => userVerificationService.VerifyUserEntity(Password, userEntity))
+                .Returns(new UserEntityVerificationResult(true, new List<int>(), new[] { 2 }, new List<int>()));
+            A.CallTo(() => userService.GetUnverifiedEmailsForUser(userEntity.UserAccount.Id))
+                .Returns(resultListingCentreEmailAsUnverified);
+            A.CallTo(() => userService.ResetFailedLoginCount(userEntity.UserAccount)).DoesNothing();

```

Comment:
Don't think you're changing anything with this line - think it can be removed.

Reply:
I think we decided that we do want to reset the failed login count if a user logs in but has an unverified email, right? I previously had that call after returning the unverified email result, but I've just moved it beforehand so this method is called now

Reply:
Yeah, but this line is just setup, and I'm pretty sure it's putting it in the default state. It's not an assertion.

Reply:
If I remove it, the test fails with this error: `Call to unconfigured method of strict fake: DigitalLearningSolutions.Web.Services.IUserService.ResetFailedLoginCount...`

Reply:
Oh, for some reason Alex made it a strict fake (line 34). I'm sure it made sense at the time. Let's leave it as is then.